lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2347,14 +2347,23 @@ llvm::Triple SwiftASTContext::GetTriple() const {
   return llvm::Triple(m_compiler_invocation_ap->getTargetTriple());
 }
 
-/// Conditions a triple string to be safe for use with Swift.  Right
-/// now this just strips the Haswell marker off the CPU name.
+/// Conditions a triple string to be safe for use with Swift.
+///
+/// This strips the Haswell marker off the CPU name (for Apple targets).
+///
+/// It also add the GNU environment for Linux.  Although this is technically
+/// incorrect, as the `*-unknown-linux` environment represents the bare-metal
+/// environment, because Swift is currently hosted only, we can get away with
+/// it.
```

Comment:
I always thought GNU was a vendor, not an environment? Is it both? Can you add to the comment what the effect of this is / why this is needed?
The aim of this function is to translate the real triple into a triple that we have a stdlib for. If that is what this is enables, I'm on board with the change.

Please cherry-pick to master-rebranch and master-next, too.

Reply:
GNU isn't a vendor, its an environment - the distributions are the vendors, and they all largely use `unknown` as they dont have any vendor extensions.  I can certainly extend the comment (most of that is in the commit message anyways).

Yeap, you nailed exactly: this is translating a triple (that is computed by lldb from `repl_swift`) into a triple that the Swift compiler can work with because it has a standard library for that target.

Ah, the joys of the rebranching, thanks for the reminder on that!

Reply:
>  An example is SuSE which vends a toolchain as x86_64-suse-linux-gnu.

Ah that makes perfect sense. Thanks for clearing that up!